tests/common/mod.rs

```diff
@@ -0,0 +1,278 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+//
+// For some reason clippy & check think that the public methods in this file are not used
+// (even though they are). Let's allow dead code so that we don't get warnings all the time.
+// This is a bit sad, hopefully it gets fixed in more recent rust versions.
+#![allow(dead_code)]
+/// This module defines common subscribers that showcase the usage of the event-manager.
+///
+/// 1. CounterSubscriber:
+///     - a dummy subscriber that increments a counter on event
+///     - only uses one event
+///     - it has to be explicitly mutated; for this reason it implements `MutEventSubscriber`.
+///
+/// 2. CounterSubscriberWithData:
+///     - a dummy subscriber that increments a counter on events
+///     - this subscriber takes care of multiple events and makes use of `Events::with_data` so
+///       that in the `process` function it identifies the trigger of an event using the data
+///       instead of the file descriptor
+///     - it has to be explicitly mutated; for this reason it implements `MutEventSubscriber`.
+///
+/// 3. CounterInnerMutSubscriber:
+///     - a dummy subscriber that increments a counter on events
```

Comment:
nit: I think ``` - a dummy subscriber that increments a counter on events``` is superfluous as it applies to all subscribers; it brings more value to mention under each subscriber type what it does differently from the others

Reply:
Hmmm, I think the differences are already documented. Do you think something more is needed? I wouldn't delete the first line with what the subscriber does because I am thinking that we might add more subscriber types in the future, and it would make sense to have a short description of them as a first comment.